kernel/kernel-api/src/main/java/io/delta/kernel/internal/metrics/TransactionMetrics.java

```diff
@@ -27,32 +30,82 @@
  */
 public class TransactionMetrics {
 
+  /** @return a fresh TransactionMetrics object with a default fileSizeHistogram (with 0 counts) */
```

Comment:
One thing to clarify: some metrics like addFileCounter etc are metrics for the commit and fileSizeHistogram (if present) is for all the files in the table. May be rename the fileSizeHistogram to tableFileSizeHistogram or related to indicate it is a table level metric?

Reply:
yeah that's a good idea to help clarify; I agree it's a little weird the two distinctions thanks